hibernate-core/src/main/java/org/hibernate/query/sqm/internal/ConcreteSqmSelectQueryPlan.java

```diff
@@ -491,7 +491,7 @@ private static CacheableSqmInterpretation buildCacheableSqmInterpretation(
 								executionContext.getQueryParameterBindings(),
 								executionContext.getSession().getLoadQueryInfluencers(),
 								sessionFactory.getSqlTranslationEngine(),
-								true
+								sqm.getCteStatements().isEmpty()
```

Comment:
We shouldn't do this for the entire query, only because there are CTEs involved. IIUC, this only matters when rendering the CTE itself. We should rather change the `BaseSqmToSqlAstConverter#deduplicateSelectionItems` to `false` before visiting the CTEs themselves (e.g. in `org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter#visitCteContainer`) like we do for subqueries.